### PR TITLE
chore: clarify various behaviours in README and cover with tests

### PR DESCRIPTION
**Pull request by jpmunz**
## Goal

Clarifying a a few different scenarios that weren't previously in the README:
* how spans that are not ended are treated
* span attribute values must be strings
* setting up console exporters for debugging
* user manager handling cases where storage APIs aren't available

### CODE REVIEW

**Review by gregorio-gerardi — APPROVED** [review]

File: src/sdk/initSDK.test.ts [hunk 1]
```diff
@@ -248,12 +251,19 @@ describe('initSDK', () => {
...
+            // This instrumentation does its own patching of Fetch which interferes with our test stub
+            '@opentelemetry/instrumentation-fetch',
+            // Document load instrumentation generates a bunch of spans in this test environment
+            '@opentelemetry/instrumentation-document-load',
+          ]),
         },
       });
       void expect(result).not.to.be.false;
 
+      // Needed to allow the browser detector resources to be grabbed
```
[the remark this answers is not in the document]
Comment (jpmunz) [reply src/sdk/initSDK.test.ts:264]: yep always async, there is an option to block on waiting for the resource attributes to populate but I think this is an edge case that only shows up in testing because we immediately end the session after initializing, in a real app there should be time for this to become populated on its own
